src/ts/content/adapters/tags/index.ts

```diff
@@ -1,18 +1,19 @@
-import {TagSearchOptions, TagTrees} from "./types";
+import {TagSearchOptions, TagNodes, TagTree} from "./types";
 import {getTagTrees} from "./getTags";
 
 const getAncestry = async (tag: string): Promise<string[]> => {
-	const trees = await getTagTrees();
+	const {nodes} = await getTagTrees();
 	const ancestry = [];
-	for (let node = trees.get(tag.toLowerCase()); node; node = node.parent) {
-		ancestry.push(node.tag);
+	for (let node: TagTree["parent"] = nodes.get(tag.toLowerCase()); node && "parent" in node; node = node.parent) {
+		// TODO maybe we should detect collisions here and then create a modal?
```

Comment:
how often does this code get run? and what would the user do with this information if we show them the modal?

Reply:
I think maybe the modal should happen at a later step.
For example, when adding ancestor tags.

Or maybe? Instead of a tree its a network, and we just go for it and add all ancestors on multiple paths?